packages/core/src/typegenTypes.ts

```diff
@@ -164,33 +172,37 @@ type AllowAllEvents = {
   eventsCausingServices: Record<string, string>;
 };
 
-export type ResolveTypegenMeta<
+export interface ResolveTypegenMeta<
   TTypesMeta extends TypegenConstraint,
   TEvent extends EventObject,
   TAction extends BaseActionObject,
   TServiceMap extends ServiceMap
-> = TTypesMeta extends TypegenEnabled
-  ? TTypesMeta & {
-      indexedActions: IndexByType<TAction>;
-      indexedEvents: MergeWithInternalEvents<
-        IndexByType<
-          | (string extends TEvent['type'] ? never : TEvent)
-          | GenerateServiceEvents<
-              TServiceMap,
-              Prop<TTypesMeta, 'invokeSrcNameMap'>
-            >
-        >,
-        Prop<TTypesMeta, 'internalEvents'>
-      >;
-    }
-  : MarkAllImplementationsAsProvided<TypegenDisabled> &
-      AllowAllEvents & {
+> {
+  '@@xstate/typegen': TTypesMeta['@@xstate/typegen'];
+  resolved: TTypesMeta extends TypegenEnabled
+    ? TTypesMeta & {
         indexedActions: IndexByType<TAction>;
-        indexedEvents: Record<string, TEvent> & {
-          __XSTATE_ALLOW_ANY_INVOKE_DATA_HACK__: { data: any };
-        };
-        invokeSrcNameMap: Record<
-          string,
-          '__XSTATE_ALLOW_ANY_INVOKE_DATA_HACK__'
+        indexedEvents: MergeWithInternalEvents<
+          IndexByType<
+            | (string extends TEvent['type'] ? never : TEvent)
+            | GenerateServiceEvents<
+                TServiceMap,
+                Prop<TTypesMeta, 'invokeSrcNameMap'>
+              >
+          >,
+          Prop<TTypesMeta, 'internalEvents'>
         >;
-      };
+      }
+    : TypegenDisabled &
+        AllImplementationsProvided &
+        AllowAllEvents & {
+          indexedActions: IndexByType<TAction>;
+          indexedEvents: Record<string, TEvent> & {
+            __XSTATE_ALLOW_ANY_INVOKE_DATA_HACK__: { data: any };
```

Comment:
Will this show up in intellisense ever?

Reply:
I dont think so - but if we catch this in any display then we can revisit this. Note that this PR didnt introduce this thing. 

My hope is to introduce tests for type displays, autocomplete etc. With such tests we could control this stuff with much better confidence